tasks/eva_2357/populate_ids.py

```diff
@@ -0,0 +1,196 @@
+    by_assigned_molecule = dict()
+    by_genbank = dict()
+    by_refseq = dict()
+    by_ucsc = dict()
+    with open(assembly_report_file, 'r') as f:
+        for line in f:
+            if line.startswith('#'):
+                continue
+
+            columns = line.strip().split('\t')
+            is_chromosome = columns[1] == 'assembled-molecule' and columns[3] == 'Chromosome'
+            synonyms = {'name': columns[0],
+                        'assigned_molecule': columns[2] if is_chromosome else None,
+                        'is_chromosome': is_chromosome,
+                        'genbank': columns[4],
+                        'refseq': columns[6],
+                        'is_genbank_refseq_identical': (columns[5] == '='),
+                        'ucsc': columns[9] if columns[9] != 'na' else None}
+
+            by_name[columns[0]] = synonyms
+            by_genbank[columns[4]] = synonyms
+            by_refseq[columns[6]] = synonyms
+
+            if synonyms['assigned_molecule'] is not None:
+                by_assigned_molecule[columns[2]] = synonyms
+
+            if synonyms['ucsc'] is not None:
+                by_ucsc[columns[9]] = synonyms
+
+    logging.info('Loaded chromosome synonyms for assembly {}'.format(assembly_accession))
+    return by_name, by_assigned_molecule, by_genbank, by_refseq, by_ucsc
+
+
+def get_genbank(synonym_dictionaries, contig):
+    """Returns a tuple (genbank, was_already_genbank) or raises an exception if the contig was not found"""
+    by_name, by_assigned_molecule, by_genbank, by_refseq, by_ucsc = synonym_dictionaries
+    if contig in by_genbank:
+        return contig, True
+    if contig in by_name:
+        return by_name[contig]['genbank'], False
+    if contig in by_assigned_molecule:
+        return by_assigned_molecule[contig]['genbank'], False
+    if contig in by_ucsc:
+        return by_ucsc[contig]['genbank'], False
+    if contig in by_refseq and by_refseq[contig]['is_genbank_refseq_identical']:
+        return by_refseq[contig]['genbank'], False
+    raise Exception(f"Could not find synonym for contig {contig}")
+
+
+def get_SHA1(variant_id):
+    """Calculate the SHA1 digest from the seq, study, contig, start, ref, and alt attributes of the variant"""
+    return hashlib.sha1(variant_id.encode()).hexdigest().upper()
+
+
+def get_ids(assembly, contig_synonym_dictionaries, variant_query_result, hash_to_variant_id):
+    """
+    Get a dictionary with the submitted variant hash as key and the variant id (chr_start_ref_alt) from the variant
+    warehouse as value
+    """
+    for file in variant_query_result['files']:
+        study = file['sid']
+        start = variant_query_result['start']
+        ref = variant_query_result['ref']
+        alt = variant_query_result['alt']
+        id_variant_warehouse = variant_query_result['_id']
+        try:
+            genbank_chr, was_already_genbank = get_genbank(contig_synonym_dictionaries, variant_query_result['chr'])
```

Comment:
We don't seem to use **was_already_genbank** anywhere?

Reply:
You are right, I removed it